core/src/validator.rs

```diff
@@ -169,6 +170,7 @@ pub struct ValidatorConfig {
     pub accounts_shrink_ratio: AccountShrinkThreshold,
     pub wait_to_vote_slot: Option<Slot>,
     pub ledger_column_options: LedgerColumnOptions,
+    pub compute_budget: Option<ComputeBudget>,
```

Comment:
Absolutely agreed, I was hoping to pipe this in without affecting the main `ValidatorConfig`, since as you say, if someone modifies this by hand and runs their validator with it, they will absolutely break consensus.  Thankfully there's no flag exposing it in the main `solana-validator` command.

As for analogues, `bpf_jit` is a pretty close analogue that also impacts the runtime.  We could bundle it with  `compute_budget` and `dev_halt_at_slot` into a `RuntimeConfig` type, which is either part of `ValidatorConfig` or passed separately into `Validator::new`.  More churn there, but it could be clearer.  What do you think?

Reply:
I like that idea! I'm not sure I have a strong opinion about putting it in `ValidatorConfig` vs passed separately into `Validator::new`... which do you prefer?

Reply:
I think there's already going to be quite a bit of churn with the new type, so let's go with including it in `ValidatorConfig` :sweat_smile: